docs/book/src/tasks/kubeadm-control-plane.md

```diff
@@ -0,0 +1,28 @@
+# Kubeadm control plane
+
+Using the Kubeadm control plane type to manage a control plane provides several ways to upgrade control plane machines.
+
+## Upgrading
+
+The high level steps to fully upgrading a workload cluster are to first upgrade the control plane and then upgrade
+the worker machines.
+
+### Upgrading workload machines managed by a `MachineDeployment`
+
+It is recommended to manage workload machines with a `MachineDeployment` as this will transparently manage `MachineSet`s
+and `Machine`s to allow for a seamless scaling experience. A modification to the `MachineDeployment`s spec will begin a
+rolling update of the workload machines.
+
+### How to upgrade the underlying machine image
+
+To upgrade the control plane machines underlying machine images, the `MachineTemplate` resource referenced by the
+`KubeadmControlPlane` must be changed. Since `MachineTemplate` resources are immutable, the recommended approach is to
+copy the existing `MachineTemplate`, modify the necessary values, create the new `MachineTemplate` on the management
+cluster, and finally, modify the existing `KubeadmControlPlane` resource to reference the new `MachineTemplate`
+resource. This will trigger a rolling update of the control plane using the new values found in the `MachineTemplate`.
+
+### How to upgrade the Kubernetes control plane version
+
+To upgrade the Kubernetes control plane version, which will likely, depending on the provider, also upgrade the
+underlying machine image, make a modification to the `KubeadmControlPlane` resource's `Version` field. This will trigger
+a rolling upgrade of the control plane.
```

Comment:
Might be good to add a statement along the lines of:

```
If the infrastructure provider used requires updating image IDs to update the Kubernetes version, the update for the MachineTemplate must be done at the same time as updating the KubeadmControlPlane.Spec.Version field to the appropriate version for the image. This is to ensure that the deployed control plane pods and additional upgrade steps, such as updating the kube-proxy daemonset for the workload cluster, will use the correct matching Kubernetes version as the on-disk binaries provided by the image.
```

Might be good to add a similar blurb related to image based overrides for the provided KubeadmConfig, but both of these can easily be done later and shouldn't necessarily be blocking for this PR.

Reply:
I added a paragraph around the infrastructure provider changes but not a section for the KubeadmConfig. 

Let's add that along with how to upgrade etcd and add ons maybe after we get this in place?